[Wu Licheng] iP

Licheng-Wu opened the pull request.

Commit:
Add Gradle support

Licheng-Wu committed:
Added Echo.java and feature to echo 1 command

Licheng-Wu committed:
Added feature where Duke does not terminate until "bye"

Licheng-Wu committed:
Updated chatbot's name to Mrs Dino

Licheng-Wu committed:
Revert "Updated chatbot's name to Mrs Dino"

This reverts commit f7774b9f3d59b629f6cc6434fb4a94e9111e2f5e.

Licheng-Wu committed:
Modified Ascii-art

Licheng-Wu committed:
Updated bot's welcome message

Licheng-Wu committed:
Added ability to store and list tasks

Licheng-Wu committed:
Updated Duke.java to initialise Level-2 on startup

Licheng-Wu committed:
Added Task.java class to encapsulate a task object

Licheng-Wu committed:
Added ability to mark tasks as done

Licheng-Wu committed:
Added Deadline, Event and Todo classes

Licheng-Wu committed:
Changed name of returnTask() to toString()

Licheng-Wu committed:
Added toString() methods to Deadline, Event and Todo

Licheng-Wu committed:
Used split string to accomodate for deadline, event and todo

Licheng-Wu committed:
Simplified code by removing the need to have multiples of the same catch block

Licheng-Wu committed:
Add ability to handle new todo

Licheng-Wu committed:
Added ability to handle new deadline and event tasks

Licheng-Wu committed:
Updated automated testing, added sc.close() to fix error

Licheng-Wu committed:
Added more inputs for automated testing

Licheng-Wu committed:
Modified to use switch statements, added custom exceptions

Licheng-Wu committed:
Added DukeIndexOutOfBoundsException

Licheng-Wu committed:
Compress exceptions into only DukeException

Licheng-Wu committed:
Updated Automated Testing

Licheng-Wu committed:
Added delete function

Licheng-Wu committed:
Added enum

Licheng-Wu committed:
Add Level-7 Save

Currently the changes to the List is not saved in the hard disk.

In order to backup the data, added functionality to create a text file to save data, as well as to load, modify and delete data from the text file.

Licheng-Wu committed:
Add Level-8 Dates and Times

Previously, user can input date in any format that they want.

Modified it such that user now has to enter dates for Deadline and Even in a specific yyyy-mm-dd format.

Licheng-Wu committed:
Merge branch 'branch-Level-7'

Licheng-Wu committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/TodoList.java

Licheng-Wu committed:
Fix bugs with Event.java

Licheng-Wu committed:
Refactor code to use more OOP

Licheng-Wu committed:
Divide classes into packages

Licheng-Wu committed:
Organise classes into packages again

Licheng-Wu committed:
Add JUnit tests

Licheng-Wu committed:
Package the App as a JAR file

Licheng-Wu committed:
Add JavaDoc comments

Comments added to Duke, TaskList and all Task-related classes

Licheng-Wu committed:
Tweak the code to comply with a coding standard

Currently, there is indentation for case clauses.

Changed intelliJ settings such that there is no indentation for case clauses, to ocmply with this module's coding standard

Licheng-Wu committed:
Add feature to allow users to find a task by searching for a keyword

Licheng-Wu committed:
Merge branch 'branch-A-JavaDoc'

Licheng-Wu committed:
Merge branch 'A-CodingStandard'

Licheng-Wu committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/TaskManager.java

Licheng-Wu committed:
Reverse commits

Licheng-Wu committed:
Modify code to follow coding standard

Licheng-Wu committed:
Merge branch 'branch-A-CodingStandard'

Licheng-Wu committed:
Merge branch 'add-gradle-support' of https://github.com/Licheng-Wu/ip into branch-A-Gradle

Licheng-Wu committed:
Merge branch 'branch-A-Gradle'

Licheng-Wu committed:
Add CheckStyle

Added the ability for gradle to checkstyle by adding checkstyle.xml and supressions.xml (taken from addressbook3)

Licheng-Wu committed:
Merge branch 'branch-A-CheckStyle'

Licheng-Wu committed:
Add Gradle wrappers

Licheng-Wu committed:
Add checkstyle

Licheng-Wu committed:
Merge branch 'branch-A-Gradle'

# Conflicts:
#	gradle/wrapper/gradle-wrapper.properties
#	gradlew
#	gradlew.bat

Licheng-Wu committed:
Merge branch 'branch-A-CheckStyle'

Licheng-Wu committed:
Rename suppressions.xml

Licheng-Wu committed:
Merge branch 'branch-A-CheckStyle'

Licheng-Wu committed:
Fix style

Licheng-Wu committed:
Merge branch 'master' of https://github.com/Licheng-Wu/ip

Licheng-Wu committed:
Add javaFX as shown in tutorial

Shows Hello World as the label in GUI

Door-oof commented on src/main/java/duke/Echo.java:
```diff
@@ -0,0 +1,23 @@
+    public static void initialise() {
+        while (!Echo.terminate) {
+            Scanner sc = new Scanner(System.in);  // Create a Scanner object
+             String command = sc.nextLine();  // Read user input
```
The indentation is not in line with the rest.

Licheng-Wu replied:
Noted and removed the extra space, thank you!

Door-oof commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,21 @@
+package duke;
+
+public class Parser {
```
Perhaps include JavaDoc and specify what the class does? This can also apply to the other classes that do not have JavaDoc written.

Licheng-Wu replied:
Noted will add JavaDoc soon

Door-oof commented on src/main/java/duke/TaskManager.java:
```diff
@@ -0,0 +1,139 @@
+        sc.close();
+    }
+
+    public void handleCommand(String command) {
```
Perhaps the code can be put in 1 try-catch block?

Door-oof commented on src/main/java/duke/TaskManager.java:
```diff
@@ -0,0 +1,139 @@
+            try {
+                Command enumCommand = Command.stringToEnum(splitCommand[0]);
+                switch (enumCommand) {
+                case LIST: {
```
Should the parenthesis after the case clause be taken out for better readability?

Door-oof commented on src/main/java/duke/Echo.java:
```diff
@@ -0,0 +1,23 @@
+import java.util.Scanner;
+
+public class Echo {
+    private static boolean terminate = false;
```
Perhaps having a more intuitive variable name to make it sound more like a boolean?

Door-oof reviewed (commented):
Overall, the code is readable but there are some parts that can maybe be improved on. It is still a nice job done.

Licheng-Wu committed:
Add basic GUI

Licheng-Wu committed:
Extract command methods into classes

Licheng-Wu committed:
Merge branch 'branch-Level-10'

# Conflicts:
#	data/data.txt

Licheng-Wu committed:
Delete unused code

Licheng-Wu committed:
Add a GUI to Mrs Dino

Licheng-Wu committed:
Modify data

nicktohzyu commented on build.gradle:
```diff
@@ -0,0 +1,67 @@
+}
+
+application {
+    mainClassName = "seedu.duke.Duke"
```
Can't do gradle-tasks-application-run currently. I believe this line should be changed to
`mainClassName = "duke.Duke"`

Licheng-Wu replied:
Noted will add data storage to gitignore soon, thank you!

Licheng-Wu replied:
Sorry what command is that? Currently "gradle test" works.

nicktohzyu commented on data/data.txt:
```diff
@@ -0,0 +1,2 @@
+T | 1 | read book
```
I'd suggest adding this to your gitignore

Licheng-Wu replied:
Noted will add data storage to gitignore soon, thank you!

nicktohzyu reviewed (commented):
Your code is easy to read and well structured. I believe majority of the styling issues have been pointed out by the other reviewer. My two main suggestion are to fix the gradle run command, and to add the data storage to gitignore. You can also commit your gitignore.

Licheng-Wu committed:
Update GUI

Licheng-Wu committed:
Modify build.gradle to set app entrypoint as Laucher.java

Previously, the entrypoint is Duke.java, hence the CLI is loaded instead of GUI.

By changing the entrypoint to Launcher.java, the GUI can be loaded.

Licheng-Wu committed:
Add Assertions

Licheng-Wu committed:
Improve code quality

Licheng-Wu committed:
Add Steams

Now able to find multiple keywords.

Licheng-Wu committed:
Fix display of Tick and Cross

Licheng-Wu committed:
Merge pull request #1 from Licheng-Wu/branch-A-Assertions

Add Assertions

Licheng-Wu committed:
Merge branch 'master' into branch-A-CodeQuality

Licheng-Wu committed:
Merge branch 'master' into branch-A-Streams

Licheng-Wu committed:
Merge pull request #2 from Licheng-Wu/branch-A-CodeQuality

Branch-A-CodeQuality

Licheng-Wu committed:
Merge branch 'master' into branch-A-Streams

Licheng-Wu committed:
Merge pull request #3 from Licheng-Wu/branch-A-Streams

Branch-A-Streams

Licheng-Wu committed:
Refactor code to save tasks to storage using taskList

Licheng-Wu committed:
Add feature to reschedule Deadline and Event tasks

Deadline and Event tasks can be rescheduled using the command:
reschedule [index of task to reschedule] [new date and time of task]

Licheng-Wu committed:
Merge branch 'branch-B-Snooze'

Licheng-Wu committed:
Update GUI message for rescheduling of tasks

Licheng-Wu committed:
Improve the GUI

Licheng-Wu committed:
Add Javadoc

Licheng-Wu committed:
Add Javadocs, fix style

Licheng-Wu committed:
Improve GUI

Licheng-Wu committed:
Update ReadMe

Licheng-Wu committed:
Update ReadME

Licheng-Wu committed:
Test display of ReadME

Licheng-Wu committed:
Resize image

Licheng-Wu committed:
Revert "Resize image"

This reverts commit b60026967d883aca1e78167c6ef84a6daa2d529e.

Licheng-Wu committed:
Resize image of addTodo for ReadME

Licheng-Wu committed:
Add a User Guide

Licheng-Wu committed:
Fix issue with GUI not responding to user input

Licheng-Wu committed:
Update README.md
